### PR TITLE
Fix some warnings being emitted by clazy[WIP]

### PR DESCRIPTION
**Pull request by Waqar144**
Lots of QString changes
Almost all the signal/slot connection is now done in the modern style.
Some constness changes

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -1648,7 +1627,7 @@ void MainWindow::loadNoteFolderListMenu() {
     int c = 0;
     Q_FOREACH(QAction* action, actions) {
             // start with the 2nd item, the first item is the menu itself
-            if (c++ > 0) {
+            if (++c > 0) {
```
Comment (pbek) [inline src/mainwindow.cpp]: Are you sure you can do this? Value of `c` will be increased before assignment.

Comment (Waqar144) [reply src/mainwindow.cpp]: ah, no. Thanks. I will revert this.

Comment (pbek) [reply src/mainwindow.cpp]: I will undo that change.

Comment (pbek) [reply src/mainwindow.cpp]: Ok, I'll wait a minute more. :grin:

Comment (Waqar144) [reply src/mainwindow.cpp]: eeh. Don't merge this yet :grimacing: 
I am preparing some more changes :grimacing:

Comment (pbek) [reply src/mainwindow.cpp]: Done?

File: src/mainwindow.h [hunk 2]
```diff
@@ -689,6 +689,13 @@ private slots:
     QActionGroup *languageGroup;
     QActionGroup *spellBackendGroup;
     bool _brokenTagNoteLinksRemoved = false;
+    const QIcon tagIcon = QIcon::fromTheme(
```
Comment (pbek) [inline src/mainwindow.h]: can you please use `_` as prefix for global variables

File: src/mainwindow.cpp (hunk 1 above)
Comment (Waqar144) [reply src/mainwindow.cpp]: Done.

File: src/mainwindow.h (hunk 2 above)
Comment (Waqar144) [reply src/mainwindow.h]: Yes, sorry I am really forgetful 😬

**Comment by pbek** [comment]
I accepted the PR, you can do the global variables in another PR. :grin:

**Pull request merged**